src/lib.rs

```diff
@@ -259,7 +278,9 @@ impl Default for MessageId {
 #[allow(clippy::large_enum_variant)]
 #[derive(Serialize, Deserialize)]
 pub enum Challenge {
+    /// Request containing the challenge.
```

Comment:
I'd suggest adding some more info
```suggestion
    /// Request from the Vault to clients containing the challenge.
```